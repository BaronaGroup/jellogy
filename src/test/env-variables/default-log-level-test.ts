import {prepareTestKarhu} from '../test-utils'

describe('default-log-level-test', () => {
  const karhuTest = prepareTestKarhu({
    defaultLogLevel: 'ERROR'
  })

  it('allows overriding default log level', karhuTest((karhu, output) => {
    karhu.context('test-context').warn('Should not show up')
    process.env.KARHU_LOG_LEVEL = 'WARN'
    karhu.context('test-context').warn('Should show up')
    process.env.KARHU_LOG_LEVEL = undefined
    expect(output.tracked).toMatchSnapshot()
  }))
})
