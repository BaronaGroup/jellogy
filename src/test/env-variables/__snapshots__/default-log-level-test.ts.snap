// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default-log-level-test allows overriding default log level 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [],
  "INFO" => Array [],
  "WARN" => Array [
    Array [
      "0 WARN test-context[93m",
      "Should show up",
      "[39m",
    ],
  ],
  "ERROR" => Array [],
}
`;
